### PR TITLE
WIP: Main e2e

### PR DESCRIPTION
**Pull request by android-dev-lxl**

### CODE REVIEW

**Review by johnshea — CHANGES_REQUESTED** [review]
Bottom left corner is incorrectly colored.

Screenshot 2023-09-08 at 1 50 59 PM

**Review by johnshea — CHANGES_REQUESTED** [review]
In landscape mode, the NavigationDrawerContent width and the reply list item padding/positioning is different depending which way the emulator is rotated.

These should be consistent.

Screenshot 2023-09-19 at 1 48 15 PM

**Comment by android-dev-lxl** [comment]
Fixed!

> In landscape mode, the NavigationDrawerContent width and the reply list item padding/positioning is different depending which way the emulator is rotated.
> 
> These should be consistent.
> 
> Screenshot 2023-09-19 at 1 48 15 PM

**Comment by johnshea** [comment]
I agree it is "fixed" but not sure this is the correct fix from a UI/Material standpoint.

The column widths for navigation and details change depending on which way you rotate the device to landscape.

If you rotate left...navigation column is wide while detail column is narrow, no change for list column.

If you rotate right...navigation column is narrow while detail column is wide, no change for list column.

Shouldn't all columns remain the same size so the app is consistent no matter which way you rotate it?

Screenshot 2023-09-25 at 12 07 03 PM

**Comment by android-dev-lxl** [comment]
Added new API

File: app/src/main/java/com/example/reply/ui/theme/Theme.kt [hunk 1]
```diff
@@ -129,3 +130,24 @@ fun ReplyTheme(
         content = content
     )
 }
+
+/**
+ * Sets up edge-to-edge for the window of this [view]. The system icon colors are set to either
+ * light or dark depending on whether the [darkTheme] is enabled or not.
+ */
+private fun setUpEdgeToEdge(view: View, darkTheme: Boolean, colorScheme: ColorScheme) {
```
Comment (johnshea) [inline app/src/main/java/com/example/reply/ui/theme/Theme.kt:138]: Function needs to be removed.

File: app/src/main/java/com/example/reply/ui/theme/Theme.kt [hunk 2]
```diff
@@ -117,9 +120,7 @@ fun ReplyTheme(
     val view = LocalView.current
     if (!view.isInEditMode) {
         SideEffect {
-            val window = (view.context as Activity).window
-            window.statusBarColor = colorScheme.primary.toArgb()
-            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
+            setUpEdgeToEdge(view, darkTheme, colorScheme)
```
Comment (johnshea) [inline app/src/main/java/com/example/reply/ui/theme/Theme.kt:123]: Code needs to be reverted.

**Review by johnshea — CHANGES_REQUESTED** [review]
- 3-button navigation bar is not colored correctly.
- Column widths are different depending on which way phone is rotated to landscape. Please review the patch I previously sent which fixes this issue.

File: app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt [hunk 3]
```diff
@@ -95,7 +100,12 @@ fun ReplyHomeScreen(
         val navigationDrawerContentDescription = stringResource(R.string.navigation_drawer)
         PermanentNavigationDrawer(
             drawerContent = {
-                PermanentDrawerSheet(Modifier.width(dimensionResource(R.dimen.drawer_width))) {
+                PermanentDrawerSheet(
```
Comment (yaraki) [inline app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:103]: `PermanentDrawerSheet` doesn't work well when its container color is customized by Modifier. Remove all `Modifier..background(MaterialTheme.colorScheme.inverseOnSurface)` and use the `drawerContainerColor` parameter like this:
```
                PermanentDrawerSheet(
                    modifier = Modifier
                        .width(dimensionResource(R.dimen.drawer_width)),
                    drawerContainerColor = MaterialTheme.colorScheme.inverseOnSurface
                ) {
                    NavigationDrawerContent(
                        selectedDestination = replyUiState.currentMailbox,
                        onTabPressed = onTabPressed,
                        navigationItemContentList = navigationItemContentList,
                        modifier = Modifier
                            .wrapContentWidth()
                            .fillMaxHeight()
                            .padding(dimensionResource(R.dimen.drawer_padding_content))
                    )
                }
```

Comment (yaraki) [reply app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:103]: And you can remove `windowInsets =` as well.

Comment (android-dev-lxl) [reply app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:99]: Thank you @yaraki

**Review by johnshea — APPROVED** [review]

**Pull request merged**